mlir/test/Dialect/LLVMIR/mem2reg.mlir

```diff
@@ -1047,3 +1025,135 @@ llvm.func @scalable_llvm_vector() -> i16 {
   %2 = llvm.load %1 : !llvm.ptr -> i16
   llvm.return %2 : i16
 }
+
+// -----
+
+// CHECK-LABEL: @smaller_store_forwarding
+// CHECK-SAME: %[[ARG:.+]]: i16
+llvm.func @smaller_store_forwarding(%arg : i16) {
```

Comment:
Do we know that little-endian is the default? On tests where that matters, I'd be in favor of also explicitly setting endianness as litte.

Reply:
Little endian is the default. Currently, MLIR's data layout does not properly convey this information, but LLVM's does. I'll try to allocate time to address this matter.